keyvalue/client-internal/src/main/java/tachyon/client/keyvalue/KeyValueStores.java

```diff
@@ -38,14 +38,24 @@
     private Factory() {} // to prevent initialization
 
     /**
-     * @return a {@link KeyValueStores} instance
+     * @return a (cached) {@link KeyValueStores} instance
      */
     public static synchronized KeyValueStores create() {
       if (sKeyValueStores == null) {
-        sKeyValueStores = new BaseKeyValueStores();
+        reset();
       }
       return sKeyValueStores;
     }
+
+    /**
+     * {@link tachyon.client.ClientContext} may be reset in different tests running in the same JVM,
+     * in this case, the cached {@link KeyValueStores} needs to be updated.
+     *
+     * @return a newly created {@link KeyValueStores}
+     */
+    public static synchronized void reset() {
```

Comment:
Ideally we could make this private and call it from a private test util since it only needs to be done in tests. If you don't want to do this as a part of this change I can take care of it along with the other context classes